crates/matrix-sdk/src/account.rs

```diff
@@ -916,6 +920,91 @@ impl Account {
+    ///
+    /// This is an unstable event, so we always default to setting the unstable
+    /// version of the event for now.
+    pub async fn set_media_previews_setting(&self, new_setting: MediaPreviews) -> Result<()> {
+        let mut media_preview_config_to_update = if let Some(unstable_media_preview_config) =
+            self.get_unstable_media_preview_config().await?
+        {
+            unstable_media_preview_config
+        } else {
+            UnstableMediaPreviewConfigEventContent::new(Default::default(), Default::default())
+        };
+        media_preview_config_to_update.media_previews = new_setting;
+        // Updating the account data
+        self.set_account_data(media_preview_config_to_update).await?;
+        // TODO: I think I should reset all the storage and perform a new local sync
+        // here but I don't know how
+        Ok(())
+    }
+
+    /// Set the avatars in invite requests display policy for the account.
+    ///
+    /// This is an unstable event, so we always default to setting the unstable
+    /// version of the event for now.
+    pub async fn set_invite_avatars_setting(&self, new_setting: InviteAvatars) -> Result<()> {
+        let mut media_preview_config_to_update = if let Some(unstable_media_preview_config) =
+            self.get_unstable_media_preview_config().await?
+        {
+            unstable_media_preview_config
+        } else {
+            UnstableMediaPreviewConfigEventContent::new(Default::default(), Default::default())
+        };
+        media_preview_config_to_update.invite_avatars = new_setting;
+        // Updating the account data
+        self.set_account_data(media_preview_config_to_update).await?;
+        // TODO: I think I should reset all the storage and perform a new local sync
+        // here but I don't know how
+        Ok(())
+    }
+
+    /// Get the current media previews display policy for the account.
+    pub async fn get_media_previews_setting(&self) -> Result<MediaPreviews> {
+        if let Some(media_preview_config) = self.get_media_preview_config().await? {
+            return Ok(media_preview_config.media_previews);
+        } else if let Some(unstable_media_preview_config) =
+            self.get_unstable_media_preview_config().await?
+        {
+            return Ok(unstable_media_preview_config.media_previews);
+        }
+        Ok(Default::default())
+    }
+
+    /// Get the current avatars in invite requests display policy for the
+    /// account.
+    pub async fn get_invite_avatars_setting(&self) -> Result<InviteAvatars> {
+        if let Some(media_preview_config) = self.get_media_preview_config().await? {
+            return Ok(media_preview_config.invite_avatars);
+        } else if let Some(unstable_media_preview_config) =
+            self.get_unstable_media_preview_config().await?
+        {
+            return Ok(unstable_media_preview_config.invite_avatars);
+        }
```

Comment:
This is tricky to change to `unwrap_or_else` style because the first else part requires `.await` and `?`, both of which don't fit inside a closure.

I would definitely prefer to remove the `return`s by making the whole function a single `if` block i.e. making the last part:

```
} else {
	Default::default()
}
```

Reply:
the problem is that the unstable and stable ruma types are different, so using a single `unwrap_or_else` can't be used.

Maybe I could implement a trait on the SDK side for both of them? But I am not sure if is worth to just remove an elif